crates/ruff_python_parser/src/lexer.rs

```diff
@@ -854,21 +852,23 @@ impl<'src> Lexer<'src> {
             }
         } else {
             if let Err(error) = self.skip_whitespace() {
-                return self.push_error_with_range(error);
+                return self.push_error(error);
             }
         }
 
+        // The lexer might've skipped whitespaces, so update the start offset
         self.cursor.start_token();
+
```

Comment:
This is the only reference which we need to keep

Reply:
I'm not sure what you mean by that. The only reference to what?

Reply:
The only reference to `self.cursor.start_token()`. Earlier, the calls were spread across multiple `lex_*` calls but now it's centralized in the `next_token`. But, this still needs to remain because the lexer might skip whitespaces before this.